src/material.cpp

```diff
@@ -469,13 +519,173 @@ void pyne::Material::write_hdf5(std::string filename, std::string datapath,
+
+  // Set file access properties so it closes cleanly
+  hid_t fapl;
+  fapl = H5Pcreate(H5P_FILE_ACCESS);
+  H5Pset_fclose_degree(fapl, H5F_CLOSE_STRONG);
+  // Create new/open datafile.
+
+  // This complicated algorithm is required to allow backward compatibility with
+  // previous version of write_hdf5 (where data were written in a hdf5 DATASET)
+  // FILE EXIST ?
+  //    NO -> create file with a "/material" group, write the data in it
+  //    YES:
+  //      - DATAPTH exist:
+  //        - YES && != /material -> Detect NUCPATH + old write_hdf5
+  //        - NO -> CONTINUE
+  //          - "/material" EXIST:
+  //            - NO -> create "/material" group and write everything in it
+  //            - YES:
+  //              - "/material" is a DATASET -> detect NUCPATH + old write_hdf5
+  //              - "/material" is a GROUP -> add data in it
+  //              - "/material" isn't a GROUP nor a DATASET -> fail and complain
+
+  hid_t db;
+  if (!pyne::file_exists(filename)) {
+    // Create the file
+    db = H5Fcreate(filename.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, fapl);
+
+  } else {
+    bool ish5 = H5Fis_hdf5(filename.c_str());
+    if (!ish5) throw h5wrap::FileNotHDF5(filename);
+    db = H5Fopen(filename.c_str(), H5F_ACC_RDWR, fapl);
+  }
+  int prot1_hdf5_layout = detect_hdf5_layout(db, mat_name);
+  switch (prot1_hdf5_layout) {
+    case prot1_layout::unknown: {
+      throw std::runtime_error(
+          mat_name +
+          " is not a dataset and /material entity is neither a group nor a dataset.");
+      break;
+    }
+    
+    // old layout
+    case prot1_layout::old_layout: {
+      std::string nucpath = "/nucid";
+      hid_t data_set = H5Dopen2(db, mat_name.c_str(), H5P_DEFAULT);
+
+      if (h5wrap::path_exists(db, mat_name)) {
+        // if path exists grab the nucpath
+        bool nucpath_detetcted = detect_nuclidelist(data_set, nucpath);
+        if (!nucpath_detetcted) {  // can't find a valid nuclide list path
+                                   // from mat_name... fail
+          throw std::runtime_error(
+              "Can't find the nuclide list path in the existing mat_name. "
+              "Can't "
+              "add your material to the mat_name.");
+        }
+      }
+      deprecated_write_hdf5(db, mat_name, nucpath, row, chunksize);
+      H5Fclose(db);
+      return;
+      break;
+    }
+
+    // no "/material or matname in the hdf5 file -> build the new layout
+    case prot1_layout::path_donotexists: {
+      material_grp_id =
+          H5Gcreate2(db, "/material", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+      break;
+    }
+
+    // /material as a group -> new layout: open the "/material" group
+    case prot1_layout::new_layout: {
+      material_grp_id = H5Gopen2(db, "/material", H5P_DEFAULT);
+      break;
+    }
+  }
+  
+
+  //mat_name is proided with a "/" so need to open with fullpath
+  // Group "/material/mat_name" does not exist create it
```

Comment:
You still have "/material" as the default value of the variable `mat_name`, so this comment should refer to "/material/material" (or change the default value?)